client/packages/coldchain/src/Equipment/ListView/AddFromScannerButton.tsx

```diff
@@ -64,9 +65,24 @@ export const AddFromScannerButtonComponent = () => {
 
   const handleScanResult = async (result: ScanResult) => {
     if (!!result.content) {
-      const { content } = result;
+      const { gs1 } = result;
 
-      const asset = await scanAsset(content).catch(() => {});
+      if (!gs1) {
+        // try to fetch the asset by id, as it could be an id from our own barcode
```

Comment:
I like the move of this to the frontend!

Reply:
It's nice now, as we have the frontend library to do the work to figure out if it's GS1.